src/fpmath.h

```diff
@@ -47,7 +47,7 @@
 #define _PDP_ENDIAN       __ORDER_PDP_ENDIAN__
 #define _BYTE_ORDER       __BYTE_ORDER__
 
-#elif defined(__linux)
+#if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
```

Comment:
@EdSchouten I'm not sure now whether ```__FreeBSD_kernel__``` is defined on real FreeBSD, but we obviously don't want to break that build.

Another option is to use ```__GLIBC__``` here which should be defined on Linux, GNU/kFreeBSD and GNU/Hurd.